### ISSUE TITLE
Allow configuring audit logs in KedaController

### ISSUE THREAD

**Issue by dgrisonnet** [kedacore/keda-olm-operator#114]
**tl;dr: KEDA 2.5.0 will have options to configure audit logs for its metrics server, so it would be great to make them customizable by the OCP users**

The KedaController custom resource currently only allows configuring Keda's log level, which will not be very helpful when trying to understand where an autoscaling request is coming from. This is key information when backends are getting overloaded by autoscaling requests made by user applications and we are trying to figure out who is the rogue application. To make that possible the [custom-metrics-apiserver](https://github.com/kubernetes-sigs/custom-metrics-apiserver) library added the capacity to configure audit logs to the generic apiserver that is serving the Kubernetes metrics API. This feature is only available starting from v1.22 of the library which is what KEDA's main branch is using, but it isn't in any official release of the project yet. Once a new official release is cut and brought downstream, I think we should make it possible for the users to enable audit logs since this information can be extremely valuable for debugging purposes.

### Use-Case

An openshift customer opens a Bugzilla saying that Prometheus has a constant 100% CPU usage. After investigating we know that the application making the most consuming requests to Prometheus is KEDA. However, we don't know which application is making the requests to KEDA. With audit logs, we could track down where the autoscaling request is coming from and identify the rogue application that is behind the requests.

A similar effort is in progress to have audit logs for the resource metrics API: https://issues.redhat.com/browse/MON-1786
